### PR TITLE
debian/rules, ceph.spec.in: invoke cmake with -DBOOST_J

### PR DESCRIPTION
**Pull request by dmick**
Allow boost build during toplevel cmake from Debian package build
to benefit from multiple processors.  Should speed build a lot
on many-proc machines (say, arm64).

Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by dmick** [comment]
The equivalent change for rpm is somewhat more complex; researching.

**Comment by dmick** [comment]
@liewegas sorry, added the rpm stuff.  need to test that it builds.

File: debian/rules [hunk 1]
```diff
@@ -11,6 +11,8 @@ extraopts += -DWITH_CEPHFS_JAVA=ON
 extraopts += -DCMAKE_INSTALL_LIBDIR=/usr/lib
 extraopts += -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib
 extraopts += -DCMAKE_INSTALL_SYSCONFDIR=/etc
+NPROC=$(shell nproc)
```
Comment (tchaikov) [inline debian/rules]: see https://www.debian.org/doc/debian-policy/ch-source.html, we might want to put something like
```sh
ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
  NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
  extraopts += -DBOOST_J=$(NUMJOBS)
endif
```

and let the pbuilder decide the parallel level of the build, currently we use "-j`grep -c processor /proc/cpuinfo`", see https://github.com/ceph/ceph-build/blob/master/ceph-build/build/build_deb#L83.

Comment (dmick) [inline debian/rules]: good call.  I'll rework tomorrow; all the lab slaves are offline for reasons I don't understand at the moment.

**Comment by dmick** [comment]
Used @tchaikov's suggestion for extracting debian parallelism, fixed missing '\', all looks well at https://jenkins.ceph.com/job/ceph-dev-new-build/2024/ and its subjobs.

**Comment by liewegas** [comment]
fwiw the earlier version of this ran ok, http://pulpito.ceph.com/sage-2017-03-24_17:15:23-rados-wip-sage-testing---basic-smithi/

**Comment by dmick** [comment]
I'm ready to accept this if y'all are.  The centos7 build on aarch64 has another problem covered in https://github.com/ceph/ceph/pull/14132 but the boost section of the build showed the higher parallelism.

**Pull request merged**